packages/mui-base/src/AutocompleteUnstyled/useAutocomplete.js

```diff
@@ -526,16 +526,19 @@ export default function useAutocomplete(props) {
     // eslint-disable-next-line react-hooks/rules-of-hooks
     React.useEffect(() => {
       if (!inputRef.current || inputRef.current.nodeName !== 'INPUT') {
-        console.error(
-          [
-            `MUI: Unable to find the input element. It was resolved to ${inputRef.current} while an HTMLInputElement was expected.`,
-            `Instead, ${componentName} expects an input element.`,
-            '',
-            componentName === 'useAutocomplete'
-              ? 'Make sure you have binded getInputProps correctly and that the normal ref/effect resolutions order is guaranteed.'
-              : 'Make sure you have customized the input component correctly.',
-          ].join('\n'),
-        );
+        const errorText = [
+          `MUI: Unable to find the input element. It was resolved to ${inputRef.current} while an HTMLInputElement was expected.`,
+          `Instead, ${componentName} expects an input element.`,
+          '',
+          componentName === 'useAutocomplete'
+            ? 'Make sure you have binded getInputProps correctly and that the normal ref/effect resolutions order is guaranteed.'
+            : 'Make sure you have customized the input component correctly.',
+        ].join('\n');
+        if (inputRef.current && inputRef.current.nodeName === 'TEXTAREA') {
+          console.warn(errorText);
```

Comment:
Let's describe the issue in more detail in case a textarea is provided.
Perhaps something along: "A textarea element was provided to Autocomplete where input was expected. This is not a supported scenario but it may work under certain conditions. A textarea keyboard navigation may conflict with Autocomplete controls (e.g. enter and arrow keys). Make sure to test keyboard navigation and add custom event handlers if necessary."

Reply:
Cool. that makes a lot of sense. I updated it.